lib/cyclonedx/report_go_dep_cyclonedx.rb

```diff
@@ -1,7 +1,21 @@
 module Cyclonedx
   class ReportGoDep < Base
+    TYPE = "golang".freeze
+    UNLOCKED_DEPENDENCY_FILE = "go.sum".freeze
```

Comment:
I think `go.mod` should be set here since go.sum contains the locked versions of the dependency? Otherwise the version of dependencies won't be included in the purl